dttools/src/jx.c

```diff
@@ -0,0 +1,320 @@
+/*
+Copyright (C) 2015- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "jx.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+struct jx_pair * jx_pair( struct jx *key, struct jx *value, struct jx_pair *next )
```

Comment:
An object's key must be a string. Probably the function should just take a `const char *`?